ext/NNlibEnzymeCoreExt/NNlibEnzymeCoreExt.jl

```diff
@@ -0,0 +1,374 @@
+module NNlibEnzymeCoreExt
+
+using NNlib
+import EnzymeCore
+using Random
+
+for (name, dataname, filtername) in ((typeof(NNlib.conv!), NNlib.∇conv_data!, NNlib.∇conv_filter!),
+                                     (typeof(NNlib.depthwiseconv!), NNlib.∇depthwiseconv_data!, NNlib.∇depthwiseconv_filter!) )
+    @eval begin
+
+function EnzymeCore.EnzymeRules.augmented_primal(config, func::EnzymeCore.Const{$name}, ::Type{RT},
```

Comment:
this block should be indented twice.

We can shorten the signature a bit:
```suggestion
function EnzymeRules.augmented_primal(config, func::EnzymeCore.Const{$name}, ::Type{RT},
```
and `using EnzimeCore.EnzymeRules` at the beginning.